src/devices/wifi/nm-device-wifi.c

```diff
@@ -2421,6 +2421,45 @@ build_supplicant_config (NMDeviceWifi *self,
 
 /*****************************************************************************/
 
+static gboolean
+wake_on_wlan_enable (NMDevice *device)
+{
+	NMSettingWirelessWakeOnWLan wowl;
+	NMSettingWireless *s_wireless;
+	const char *password = NULL;
+	gs_free char *value = NULL;
+
+	s_wireless = (NMSettingWireless *) nm_device_get_applied_setting (device, NM_TYPE_SETTING_WIRELESS);
+	if (s_wireless) {
+		wowl = nm_setting_wireless_get_wake_on_wlan (s_wireless);
+		password = nm_setting_wireless_get_wake_on_wlan_password (s_wireless);
+		if (wowl != NM_SETTING_WIRELESS_WAKE_ON_WLAN_DEFAULT)
+			goto found;
+	}
+
+	value = nm_config_data_get_connection_default (NM_CONFIG_GET_DATA,
+	                                               "wifi.wake-on-wlan",
+	                                               device);
+
+	if (value) {
+		wowl = _nm_utils_ascii_str_to_int64 (value, 10,
+		                                    NM_SETTING_WIRELESS_WAKE_ON_WLAN_NONE,
+		                                    G_MAXINT32,
+		                                    NM_SETTING_WIRELESS_WAKE_ON_WLAN_DEFAULT);
+
+		if (   NM_FLAGS_ANY (wowl, NM_SETTING_WIRELESS_WAKE_ON_WLAN_EXCLUSIVE_FLAGS)
+		    && !nm_utils_is_power_of_two (wowl)) {
```

Comment:
This is the same check as in `verify()`. Same comment about invalid flags applies, and what happens if MAGIC is present (but obviously, no password).